src/actors/workers/state_tracing.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of substrate-archive.
+
+// substrate-archive is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// substrate-archive is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with substrate-archive.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::error::Result;
+use sc_tracing::{SpanDatum, TraceEvent, TraceHandler, ProfilingSubscriber};
+use xtra::prelude::*;
+
+const TRACE_TARGETS: &str = "assets,atomic-swap,aura,authority-discovery,authorship, \
```

Comment:
How do we obtain this list? Can we do anything smart here? Can we stick them in a toml file at least? Ideally we should have an RPC to ask the node about the supported targets.